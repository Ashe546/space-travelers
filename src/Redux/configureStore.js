import { configureStore } from '@reduxjs/toolkit';

import rocketReducer from './rocket/rocket';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },
});

export default store;
