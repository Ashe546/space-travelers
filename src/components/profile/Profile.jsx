import { useSelector } from 'react-redux';

import MyRockets from './MyRockets';
import MyMissions from './MyMissions';
import styles from './Profile.module.css';

function Profile() {
  const rockets = useSelector((state) => state.rockets).filter(
    (rocket) => rocket.reservation,
  );

  return (
    <div className={styles.container}>
      <MyMissions />
      <MyRockets data={rockets} />
    </div>
  );
}

export default Profile;
