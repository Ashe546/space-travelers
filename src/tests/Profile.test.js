import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Profile from '../components/profile/Profile';

const mockStore = configureStore([]);

describe('Profile', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rockets: [
        {
          id: 1,
          name: 'Falcon 1',
          description: 'Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
          image: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
          reservation: false,
        },
      ],
    });

    component = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
  });

  it('should render rocket components correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
