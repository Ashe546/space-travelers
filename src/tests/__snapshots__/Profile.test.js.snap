// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render rocket components correctly 1`] = `
<div
  className="container"
>
  <div
    className="myMission"
  >
    <h2>
      My Missions
    </h2>
    <ul>
      <li>
        SpaceX
      </li>
      <li>
        SpaceX
      </li>
      <li>
        SpaceX Fal
      </li>
    </ul>
  </div>
  <div
    className="myRocket"
  >
    <h2>
      My Rockets
    </h2>
    <ul />
  </div>
</div>
`;
